airflow/www/views.py

```diff
@@ -1912,7 +1912,7 @@ def encode_ti(task_instance: Optional[models.TaskInstance]) -> Optional[List]:
                 # round to seconds to reduce payload size
                 task_instance_data[2] = int(task_instance.start_date.timestamp())
                 if task_instance.duration is not None:
-                    task_instance_data[3] = int(task_instance.duration)
+                    task_instance_data[3] = float(task_instance.duration)
```

Comment:
So, casting this to an int was an optimization for large/huge dags: https://github.com/apache/airflow/pull/7492, particularly https://github.com/apache/airflow/pull/7492#discussion_r384158218

So we might nee to make this smarter, say, and only say make it an int about 10 or 30s, and in other cases make it a float with 3dp (i.e. 0.123s, rather than the default which will be to 6 or 9 dp)

Reply:
good point.